resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -745,15 +739,20 @@ private[spark] class Client(
       // Save the YARN configuration into a separate file that will be overlayed on top of the
       // cluster's Hadoop conf.
       confStream.putNextEntry(new ZipEntry(SPARK_HADOOP_CONF_FILE))
-      yarnConf.writeXml(confStream)
+      hadoopConf.writeXml(confStream)
       confStream.closeEntry()
 
       // Save Spark configuration to a file in the archive.
       val props = new Properties()
       sparkConf.getAll.foreach { case (k, v) => props.setProperty(k, v) }
       // Override spark.yarn.key to point to the location in distributed cache which will be used
       // by AM.
-      Option(amKeytabFileName).foreach { k => props.setProperty(KEYTAB.key, k) }
+      Option(amKeytabFileName).foreach { k =>
+        // Do not propagate the app's secret using the config file.
+        if (k != SecurityManager.SPARK_AUTH_SECRET_CONF) {
```

Comment:
Is it necessary to add a check here? I'm not sure how this could happen.

Reply:
Oh, I think this is the wrong place for the check.